### PR TITLE
Add rubocop-rails and rubocop-rspec

### PR DESCRIPTION
**Pull request by kalashnikovisme**
## What's changed basically?

Just added `rubocop-rails` and `rubocop-rspec`

I use `rubocop-rspec` first time in my life, so I want to talk about its configuration with you.

Let's talk about `.rubocop.yml` according to `rubocop-rspec` gem. I added comments to every point of `rubocop-rspec` configuration.

```yaml
# One exception tests sounds good, but in practise it becomes a huge problem. Test examples run too many times and the whole test process becomes slow.
# Moreover, remember about `and` function that we can use. `expect(page).to have_css(css_1).and have_css(css_2)`
RSpec/MultipleExpectations:
  Max: 3

# Let's see at this file https://github.com/Purple-Magic/tramway/blob/main/spec/helpers/navbar_helper_spec.rb
# 3 nested groups is not enough, cause we have levels `className` -> `functionName` -> `successOrFailure` -> `certain_case`. It should be 4 at least
RSpec/NestedGroups:
  Max: 4

# According to expectations count. Let's imagine that we need 3 expectations in one test and also we need one line to make `service.call` or stuff like that. 3 * 3 + 1 = 10
RSpec/ExampleLength:
  Max: 10


# I love this kind of expression `expect(form_class).to receive(:new).with(object).and_return(form_object)`, because it's short
RSpec/StubbedMock:
  Enabled: false
```

### CODE REVIEW

**Comment by haukot** [comment]
I don't see much harm in this. A lesser value of `RSpec/ExampleLength` and `RSpec/NestedGroups` could force you to make more helpers and e.g. example groups, which _could_ increase the readability of the code. 
But I'm not a big fan of too small tests either, can't comment on that with certainty.

**Comment by kalashnikovisme** [comment]
> I don't see much harm in this. A lesser value of `RSpec/ExampleLength` and `RSpec/NestedGroups` could force you to make more helpers and e.g. example groups, which _could_ increase the readability of the code. But I'm not a big fan of too small tests either, can't comment on that with certainty.

@haukot what about helpers and that kind of stuff, there are other cops that make us use them. So, example is used only for the purpose I described in the comment in the config file.

File: spec/forms/normalizes_form_spec.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
     context 'when normalizing attributes with apply_to_nil: true' do
       it 'applies normalization to nil values and saves attributes' do
-        expect(nil).to receive(:strip).twice
+        allow(nil).to receive(:strip).twice
```
Comment (haukot) [inline spec/forms/normalizes_form_spec.rb:30]: will this still check that `strip` should be called twice?

**Comment by haukot** [comment]
The forcing of `described_class` is great, `instance_double` and `class_double` instead of `double` are good too. Not sure about `subject` replacing.
Other changes I don't see as important

File: spec/forms/normalizes_form_spec.rb (hunk 1 above)
Comment (kalashnikovisme) [reply spec/forms/normalizes_form_spec.rb:30]: Yeah!

This code `allow(nil).to receive(:strip).once` returns error in this test.

**Comment by kalashnikovisme** [comment]
@haukot I thought about subject replacing, looks like `subject` without a name should be used in cases of using `it_expected` methods only. Otherwise, the `subject` should have a name.

File: spec/forms/normalizes_form_spec.rb (hunk 1 above)
Comment (haukot) [reply spec/forms/normalizes_form_spec.rb:30]: I think I prefer `expect` syntax more as I more want to see in tests what is expected, than I want to see mocks.

**Pull request merged**